src/api-binder/report.ts

```diff
@@ -141,21 +144,45 @@ export async function startReporting() {
 	])) as StateReportOpts;
 
 	let reportPending = false;
-	const doReport = async () => {
+	// Reports current state if not already sending and does not exceed report
+	// frequency. Returns true if sent; otherwise false.
+	const doReport = async (): boolean => {
```

Comment:
```suggestion
	const doReport = async (): Promise<boolean> => {
```

Reply:
Funny what you learn when you compile things. ;-)